packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -160,22 +160,19 @@ export class DataStores implements IDisposable {
         }
 
         const flatBlobs = new Map<string, string>();
-        let flatBlobsP = Promise.resolve(flatBlobs);
-        let snapshotTreeP: Promise<ISnapshotTree> | null = null;
+        let snapshotTree: ISnapshotTree | null = null;
         if (attachMessage.snapshot) {
-            snapshotTreeP = buildSnapshotTree(attachMessage.snapshot.entries, flatBlobs);
-            // flatBlobs' validity is contingent on snapshotTreeP's resolution
-            flatBlobsP = snapshotTreeP.then(() => { return flatBlobs; });
+            snapshotTree = buildSnapshotTree(attachMessage.snapshot.entries, flatBlobs);
```

Comment:
we have the snapshot synchronously here, so by not calculating SHA, there is no need to go async